### PR TITLE
feat(search): add search element

### PR DESCRIPTION
**Pull request by ami**
(╭☞´ิ∀´ิ)╭☞

fixes https://github.com/pokedextracker/pokedextracker.com/issues/18

@robinjoseph08 added style and component bases, functionality ready to be added! 🎉 

### TODO

- [x] add search functionality
- [x] fix weird lag when you delete entire query
- [x] add empty state
- [x] fix "clear search results" in empty state
- [x] scroll to top when you type

### CODE REVIEW

File: app/components/search-bar.jsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export function SearchBarComponent () {
+  return (
+    <div className="dex-search-bar">
+      <div className="wrapper">
+        <div className="form-group">
+          <i className="fa fa-search" />
+          <input className="form-control" name="search" id="search" type="text" placeholder="Search by Pokémon name.." autoComplete="off" autoCorrect="off" autoCapitalize="off" spellCheck="false" />
```
Comment (robinjoseph08) [inline app/components/search-bar.jsx]: is this supposed to be `name..` with just 2 periods?

**Comment by robinjoseph08** [comment]
@ami added core functionality! started a todo list of things that need to be done before merge. feel free to add/remove!

**Comment by ami** [comment]
@robinjoseph08 looks good other than the things you mentioned! do you want me to add in an empty state and you can do the rest?

**Comment by robinjoseph08** [comment]
@ami yup, sounds good!

**Comment by ami** [comment]
@robinjoseph08 empty state added! comments appreciated. 2 things:

1. i thought it'd be nice to add a link to clear the search results from the empty state since that's probably where the user would be looking. while i got the action to work to clear the actual query, i couldn't figure out how to clear the text box. can you look into that? or if it's not worth it, we can just remove the link completely.

2. noticed while working on this, is if you put in a search query, then go to the user's profile, and click a different dex, the search query is retained. i don't think this is desired behavior. maybe we should clear the query every time a dex is loaded? what do you think?

**Comment by ami** [comment]
@robinjoseph08 fixed -- and all functionality lgtm now!

**Comment by robinjoseph08** [comment]
@ami added the keyboard shortcut! feel free to edit the placeholder text as you see fit

File: app/components/search-bar.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    if (this._search) {
+      this._search.value = query;
+    }
+
+    return (
+      <div className="dex-search-bar">
+        <div className="wrapper">
+          <div className="form-group">
+            <i className="fa fa-search" />
+            <input className="form-control" ref={(c) => this._search = c} name="search" id="search" type="text" placeholder="Search by Pokémon name (press / to focus)" autoComplete="off" autoCorrect="off" autoCapitalize="off" spellCheck="false" onChange={(e) => setQuery(e.target.value)} />
```
[the remark this answers is not in the document]
Comment (robinjoseph08) [reply app/components/search-bar.jsx]: image

the quotes kinda look weird? i also think "auto select" is confusing, but maybe we can just say "search" if "focus" is confusing?

[the remark this answers is not in the document]
Comment (ami) [reply app/components/search-bar.jsx]: would be cool if we could do a little key-icon thing like this.. but oh well. let's  remove the quotes then.

maybe "quick search" or "highlight" or "jump"? idk everything's weird lol

[the remark this answers is not in the document]
Comment (robinjoseph08) [reply app/components/search-bar.jsx]: not sure what "like this" is referring to, but ill remove quotes lol

ill just do "quick search". that seems like the best option ¯\\\_(ツ)_/¯

[the remark this answers is not in the document]
Comment (robinjoseph08) [reply app/components/search-bar.jsx]: o now i see

**Comment by ami** [comment]
yeyyy les mergee

**Pull request merged**